tests/user/test_user_debug_logging.py

```diff
@@ -41,7 +42,6 @@ def test_add_user_request_debug_log_works_on_success(
             )
         success_log = self.ansi_escape.sub("", stdout.getvalue())
         self.assertEqual(success_log, TestUserConstants.TEST_ADD_USER_SUCCESS_LOG)
-        self.assertNotIn(self.test_password, success_log)
```

Comment:
I think it is good to have the `self.assertNotIn()` assertions. My concern was that with those "PASSWORD_SIMPLE" tests is that the assertions on those tests added a lot of noise and in my opinion. A good assertion for those tests could be to maybe assert that you have the expected number of occurrences of `********` in debug logs and result dictionaries and that the password exists in the output to verify that only the sections that contain the actual password were redacted and that XML tags, messages, and dictionary keys that happen to contain the same string are not.

Reply:
I brought back many of the previous assertions with the exception of making the described changes to debug logging. I think that counting the "********" occurrences is a novel way of tracking the "simple" password redaction in that case. I do think, though, that there isn't really a clearer way to do this in result output. For one, SMO settings will already "redact" the password, so you have to ensure that neither the unredacted password or "redacted" password appear in the output, then also ensure that a "simple" password appears elsewhere in the output. The other hang-up is that all of these items only ever appear in the "message" portion of the result dictionary anyway, so looking in specific sections is akin to just checking if an entire string matches which the first assertion really is already doing, but I like the redundancy of specific checks. I am open to discussing this further though.